lib/AST/Module.cpp

```diff
@@ -1247,6 +1247,44 @@ bool ModuleDecl::isSystemModule() const {
   return false;
 }
 
+namespace {
+enum class SerializedModuleKind {
+  Other,
+  Clang,
+  Swift,
+};
+}
+static SerializedModuleKind getSerializedModuleKind(const ModuleDecl *M) {
+  Optional<SerializedModuleKind> K;
+
+  for (auto File : M->getFiles()) {
+    switch (File->getKind()) {
+      case FileUnitKind::Source:
+      case FileUnitKind::Builtin:
+      case FileUnitKind::Derived:
+        return SerializedModuleKind::Other;
+      case FileUnitKind::SerializedAST:
+        assert(!K.hasValue() && "cannot handle multi-file modules");
```

Comment:
I'd really prefer if you didn't ever try to ask "is this a serialized Swift module". It interferes with the idea of someday doing partial rebuilds, or having a non-precompiling script-like import-from-source.